src/parsers/manifest/dash/indexes/base.ts

```diff
@@ -198,6 +213,17 @@ export default class BaseRepresentationIndex implements IRepresentationIndex {
     return getInitSegment(this._index);
   }
 
+  /**
+   * Construct index Segment.
+   * @returns {Object}
+   */
+  getIndexSegment() : ISegment|null {
+    if (this._hasValidatedIndexSegment) {
```

Comment:
:/ So `getIndexSegment` decide whether I should need it or not?
That's not what I would expect from a function called `getIndexSegment`: to just return me the index segment.

If we want to implement it but still want to be able to not communicate it if already "validated", we might have to add a `isIndexValidated` function or something like that